```{r}
rm(list = ls())
library(Rsubread)
library(readxl)
library(vsn)
library(tsne)
library(reshape2)
library(xlsx)
library(ggpubr)
library(knitr)
library(gridExtra)
library(DESeq2)
library(ggplot2)
library(dplyr)
library(ggrepel)
library(PCAtools)
library(ggalt)
library(cowplot)
library(org.Hs.eg.db)
library(AnnotationDbi)
library(ReactomePA)
library(ReactomeContentService4R)
library(tidyverse)
library(enrichplot)
library(DOSE)
library(pheatmap)
library(clusterProfiler)
library(msigdbr)
library(annotate)
```

```{r}
H_t2g <- msigdbr(species = "Homo sapiens", category = "H") %>% dplyr::select(gs_name, entrez_gene)
#C2_t2g <- msigdbr(species = "Homo sapiens", category = "C2") %>% dplyr::select(gs_name, entrez_gene)
#C3_t2g <- msigdbr(species = "Homo sapiens", category = "C3") %>% dplyr::select(gs_name, entrez_gene)
#C5_t2g <- msigdbr(species = "Homo sapiens", category = "C5") %>% dplyr::select(gs_name, entrez_gene)
#C6_t2g <- msigdbr(species = "Homo sapiens", category = "C6") %>% dplyr::select(gs_name, entrez_gene)
msigdb <- bind_rows(H_t2g)
```

```{r}
validated_genes <- c("ICAM1", "ICAM2", "MMP2", "ITGA2", "CCL3L3", "CX3CL1", "CCL24",
                     "NDP", "DACT2", "KLF8", "RNF43", "SERPINA5", "FGFR2", "BRD4", "TNIK",
                     "CD54", "CD102")
```

```{r}
subread.counts <- readRDS("subreadcounts.rds")
```

```{r}
rawtable <- read.csv("rawtable.csv")
rcounts <- rawtable[,-1]
rownames(rcounts) <- rawtable[,1] 
```

```{r}
cnt <- rcounts[(rowSums(rcounts)>0),] 
colnames(cnt) <- sub("Aligned.out.bam", "", colnames(cnt))
colnames(cnt) <- sub("_R1_001.fastq.gz", "", colnames(cnt))
colnames(cnt) <- sub("X16140.00", "", colnames(cnt))
dim(cnt)
head(cnt)
```
```{r}
human_biotype <- read.csv("/Users/GoksuAvar/Desktop/TNIK Project/GCF_000001405.39_GRCh38.p13_feature_table.tsv", row.names = NULL, sep="\t")
```

```{r}
coding_genes <- human_biotype %>%
  dplyr::filter(class == "protein_coding" | class == "with_protein")
#write.xlsx(coding_genes, "Coding Genes List.xlsx")#, overwrite=TRUE)
```

```{r}
cnt <- subset(cnt, rownames(cnt) %in% coding_genes$symbol)
```

```{r}
geneLength <- subread.counts$annotation %>%
  dplyr::select("GeneID", "Length")
```


```{r}
coldata <- read.csv2("coldata.csv")
order_cols <- coldata$Sample_ID
cnt <-  cnt[,order_cols]
coldata <- as.matrix(coldata, rownames.force = NA)
rownames(coldata) <- coldata[,1]
dim(cnt)
head(cnt)
```
```{r}
all(rownames(coldata) %in% colnames(cnt))
```

```{r}
all(rownames(coldata) == colnames(cnt))
```


```{r}
dds <- DESeqDataSetFromMatrix(countData = cnt,
                              colData = coldata,
                              design = ~Sample_Group)
dds
```

```{r}
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]
```

```{r}
dds<- DESeq(dds)
saveRDS(dds, file = "dds.RDS")
```

```{r}
resultsNames(dds)
```

```{r}
vsd <- vst(dds, blind=FALSE)
vsdMat <- assay(vsd)
```

```{r}
vsdmelted <- melt(vsdMat)

pdf("VST.pdf")
ggplot(vsdmelted, aes(x=Var2, y=value))+
  geom_boxplot()+
  coord_flip()+
  ylab("Variance stabilized counts")+
  xlab("Sample")+
  ggtitle("VST")
dev.off()
```

```{r}
rlog <- rlog(dds)
rlogMat <- assay(rlog)
rlogmelted <- melt(rlogMat)

pdf("rlog.pdf")
ggplot(rlogmelted, aes(x=Var2, y=value))+
  geom_boxplot()+
  coord_flip()+
  ylab("rlog transformed counts")+
  xlab("Sample")+
  ggtitle("rlog")
dev.off()
```

```{r}
pdf(file="PCA_rlog.pdf", height=6, width=8)
pcaData <- plotPCA(rlog, intgroup=c("Sample_Group"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
ggplot(pcaData, aes(PC1, PC2, color=Sample_Group)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  theme(axis.title = element_text(size = 10, face = "bold"))+
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  theme(axis.title = element_text(size = 10, face = "bold"))+
  ggtitle("PCA") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line(), plot.title = element_text(size=15, face="bold")) +
  geom_text_repel(aes(label=name), size=3, color="black", nudge_y = 1, fontface="bold",  min.segment.length = unit(0, "lines"), max.overlaps = Inf)+
  #scale_color_manual(values=c("deeppink3", "yellowgreen", "red3", "darkorange1"))
  coord_fixed()
dev.off()
```
```{r}
pdf(file="PCA_vst.pdf", height=6, width=8)
pcaData <- plotPCA(vsd, intgroup=c("Sample_Group"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
ggplot(pcaData, aes(PC1, PC2, color=Sample_Group)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  theme(axis.title = element_text(size = 10, face = "bold"))+
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  theme(axis.title = element_text(size = 10, face = "bold"))+
  ggtitle("PCA") +
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line(), plot.title = element_text(size=15, face="bold")) +
  geom_text_repel(aes(label=name), size=3, color="black", nudge_y = 1, fontface="bold",  min.segment.length = unit(0, "lines"), max.overlaps = Inf)+
  #scale_color_manual(values=c("deeppink3", "yellowgreen", "red3", "darkorange1"))+
  #geom_encircle(data=subset(pcaData, Sample_Group=="KO"))+
  #geom_encircle(data=subset(pcaData, Sample_Group=="KOTGFB"))+
  #geom_encircle(data=subset(pcaData, Sample_Group=="WTTGFB"))+
  #geom_encircle(data=subset(pcaData, Sample_Group=="WT"))+
  coord_fixed()
dev.off()
```

```{r}
res_ko_wt <- DESeq2::results(dds, contrast=c("Sample_Group","KO5","WT"), alpha = 0.05)
summary(res_ko_wt)
```

```{r}
ko_wt_list<-data.frame(gene = rownames(res_ko_wt), res_ko_wt,
stringsAsFactors = F) %>% na.omit() %>% dplyr::select("log2FoldChange", "padj") %>% dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp) %>% rownames_to_column(var = "gene")

ko_wt_list <- ko_wt_list %>% 
  dplyr::mutate(padj = ifelse(ko_wt_list$padj == 0, .Machine$double.xmin, padj)) %>%
  dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp)

ko_wt_sorted <- ko_wt_list %>% 
  arrange(desc(abs(LFCminuslogp)))

write.xlsx(ko_wt_sorted, "ko_wt_sorted.xlsx")

pdf(file="Volcano_ko_wt.pdf", width=24, height=21)
ggplot(ko_wt_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2.25,color="grey", alpha=0.5)+
  geom_point(data=subset(ko_wt_list, minuslogp>(-log10(0.001))), size=2.25,color="darkorange1", alpha=1)+
  geom_point(data=subset(ko_wt_list, minuslogp>(-log10(0.001)) & abs(log2FoldChange)>(1)), size=2.25,color="red3", alpha=1)+
  geom_point(data=subset(ko_wt_list, minuslogp<(-log10(0.001)) & abs(log2FoldChange)>(1)), size=2.25,color="yellowgreen", alpha=1)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  ggtitle("TNIK Full KO vs WT")+
  theme(plot.title = element_text(size = 60, face = "bold"))+
  xlab("Log2FoldChange")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = 1, linetype="dashed")+
  geom_vline(xintercept = -1, linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  geom_label_repel(data=subset(ko_wt_list, minuslogp>(-log10(0.001)) | abs(log2FoldChange)>3), aes(label=gene), size=4, color="black", fontface="bold",  min.segment.length = unit(0, "lines"), max.overlaps = 10)
dev.off()

pdf(file="Volcano_ko_wt_validated.pdf", width=24, height=21)
ggplot(ko_wt_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2.25,color="grey", alpha=0.5)+
  geom_point(data=subset(ko_wt_list, minuslogp>(-log10(0.001))), size=2.25,color="darkorange1", alpha=1)+
  geom_point(data=subset(ko_wt_list, minuslogp>(-log10(0.001)) & abs(log2FoldChange)>(1)), size=2.25,color="red3", alpha=1)+
  geom_point(data=subset(ko_wt_list, minuslogp<(-log10(0.001)) & abs(log2FoldChange)>(1)), size=2.25,color="yellowgreen", alpha=1)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  ggtitle("TNIK Full KO vs WT")+
  theme(plot.title = element_text(size = 60, face = "bold"))+
  xlab("Log2FoldChange")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = 1, linetype="dashed")+
  geom_vline(xintercept = -1, linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  geom_label_repel(data=subset(ko_wt_list, gene %in% validated_genes), aes(label=gene), size=6, color="black", nudge_y = 0.5, fontface="bold",  min.segment.length = unit(0, "lines"), max.overlaps = 20)
dev.off()
```

```{r}
#make a named vector of P adjusted
ko_wt_gsea <- ko_wt_list$padj 
names(ko_wt_gsea) <- ko_wt_list$gene
ensgEntrez_ko_wt_gsea <- AnnotationDbi::select(org.Hs.eg.db, keys=names(ko_wt_gsea), columns='ENTREZID', keytype='SYMBOL')
df_ko_wt_gsea <- ensgEntrez_ko_wt_gsea %>%
  dplyr::inner_join(ko_wt_list, by=c("SYMBOL"="gene")) %>% 
  dplyr::select(ENTREZID,log2FoldChange,padj) %>% 
  dplyr::group_by(ENTREZID) %>%
  dplyr::summarise(lfc=mean(log2FoldChange), up=min(padj))
ko_wt_lfc<-df_ko_wt_gsea$lfc 
names(ko_wt_lfc)<-df_ko_wt_gsea$ENTREZID

y_ko_wt_reac <- gsePathway(sort((ko_wt_lfc), decreasing=T), organism = "human", pvalueCutoff=0.05, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_ko_wt_reac  <- setReadable(y_ko_wt_reac , 'org.Hs.eg.db', 'ENTREZID')
ko_wt_reac  <- y_ko_wt_reac@result
ko_wt_reac  <- ko_wt_reac  %>% arrange(desc(NES)) %>% mutate(Source = "Reactome")


y_ko_wt_kegg <- gseKEGG(sort((ko_wt_lfc), decreasing=T), organism = "hsa", pvalueCutoff=0.05, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_ko_wt_kegg  <- setReadable(y_ko_wt_kegg , 'org.Hs.eg.db', 'ENTREZID')
ko_wt_kegg  <- y_ko_wt_kegg@result
ko_wt_kegg  <- ko_wt_kegg  %>% arrange(qvalues) %>% mutate(Source = "KEGG")


y_ko_wt_gobp <- gseGO(sort((ko_wt_lfc), decreasing=T), OrgDb=org.Hs.eg.db,
                            ont   = "BP", pvalueCutoff=0.05, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_ko_wt_gobp <- clusterProfiler::simplify(y_ko_wt_gobp, by="qvalues")
y_ko_wt_gobp <- setReadable(y_ko_wt_gobp, 'org.Hs.eg.db', 'ENTREZID')
ko_wt_gobp <- y_ko_wt_gobp@result
ko_wt_gobp <- ko_wt_gobp %>% arrange(qvalues) %>% mutate(Source = "GO:BP")

y_ko_wt_gocc <- gseGO(sort((ko_wt_lfc), decreasing=T), OrgDb=org.Hs.eg.db,
                            ont   = "CC", pvalueCutoff=0.05, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_ko_wt_gocc <- clusterProfiler::simplify(y_ko_wt_gocc, by="qvalues")
y_ko_wt_gocc <- setReadable(y_ko_wt_gocc, 'org.Hs.eg.db', 'ENTREZID')
ko_wt_gocc <- y_ko_wt_gocc@result
ko_wt_gocc <- ko_wt_gocc %>% arrange(qvalues) %>% mutate(Source = "GO:CC")
ko_wt_gocc_simp <- clusterProfiler::simplify(y_ko_wt_gocc, by="qvalues")


y_ko_wt_go <- gseGO(sort((ko_wt_lfc), decreasing=T), OrgDb=org.Hs.eg.db,
                            ont   = "ALL", pvalueCutoff=0.05, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_ko_wt_go <- setReadable(y_ko_wt_go, 'org.Hs.eg.db', 'ENTREZID')


y_ko_wt_msigdb <- GSEA(sort((ko_wt_lfc), decreasing=T), pvalueCutoff=0.05, pAdjustMethod="BH", verbose=F, TERM2GENE = msigdb, eps = 1e-100)
y_ko_wt_msigdb <- setReadable(y_ko_wt_msigdb, 'org.Hs.eg.db', 'ENTREZID')
ko_wt_msigdb <- y_ko_wt_msigdb@result
ko_wt_msigdb <- ko_wt_msigdb %>% arrange(qvalues) %>% mutate(Source = "MSigDB")


saveRDS(y_ko_wt_msigdb, "msigdb_results.RDS")
saveRDS(y_ko_wt_go, "go_results.RDS")
saveRDS(y_ko_wt_reac, "reactome_results.RDS")


ko_wt_gsea_results_go <- merged_results_go@compareClusterResult %>% arrange(qvalues)
write.xlsx2(ko_wt_gsea_results_go, "ko_wt_gsea_results_go.xlsx")

ko_wt_gsea_results_others <- merged_results_others@compareClusterResult %>% arrange(qvalues)
write.xlsx2(ko_wt_gsea_results_others, "ko_wt_gsea_results_others.xlsx")

pdf(file="Gsea_volcano_ko_wt.pdf", width=15, height=15)
ggplot(ko_wt_gsea_results, aes(x=NES, y=(-log10(qvalues))))+
  geom_point(aes(size=setSize),color="grey", alpha=0.5)+
  scale_x_continuous(limits = c(-3,3), expand = expansion(mult = 0.5))+
  geom_label_repel(data=subset(ko_wt_gsea_results, NES>1.5 & qvalues < 0.01), aes(label=Description), force_pull=0, force=5, nudge_x = 3, direction = "y", hjust = 0, segment.size = 0.2)+
  geom_label_repel(data=subset(ko_wt_gsea_results, NES<(-1.5) & qvalues < 0.01), aes(label=Description), force=5, nudge_x = -3, hjust = 1, direction = "y", segment.size = 0.2)
dev.off()


msigdb_hallmarks_list_ko_wt <- c("TGFB_UP.V1_UP",
                             "HALLMARK_TGF_BETA_SIGNALING",
                             "HALLMARK_TNFA_SIGNALING_VIA_NFKB")

geneSetID_ko_wt_msigdb = match(msigdb_hallmarks_list_ko_wt,
                              y_ko_wt_msigdb@result[["Description"]])

pdf("gseaplot2_ko_wt_msigdb_hallmarks.pdf", width=12, height=10)
gseaplot2(y_ko_wt_msigdb, geneSetID = geneSetID_ko_wt_msigdb[!is.na(geneSetID_ko_wt_msigdb)], 
          pvalue_table = FALSE, 
          title="TNIK KO vs WT Selected MSigDB Signatures", 
          base_size = 18,
          rel_heights = c(2.5,0.3,0.5))+
  theme(axis.text = element_text(size=12, face = "bold"),
        axis.title = element_text(size=16))
dev.off()


go_list_ko_wt <- c("lymphocyte chemotaxis", "response to interferon-gamma",
                "granulocyte migration", "lymphocyte migration", "positive regulation of leukocyte migration", "neutrophil migration", "eosinophil migration")

pdf("heatplot_ko_wt_msigdb_hallmarks.pdf", width = 60, height=6)
enrichplot::heatplot(y_ko_wt_msigdb, 
                     showCategory = msigdb_hallmarks_list_ko_wt, 
                     foldChange = ko_wt_lfc)+
  ggtitle("TNIK KO vs WT Selected MSigDB Hallmarks")+
  theme(axis.text.y=element_text(size=12, face = "bold"),
        axis.text.x=element_text(size=9, face="bold"),
        plot.title = element_text(size = 24, face="bold"))
dev.off()


# pdf("heatplot_ko_wt_go.pdf", width = 20, height=4)
# enrichplot::heatplot(y_ko_wt_go, 
#                      showCategory = go_list_ko_wt, 
#                      foldChange = ko_wt_lfc)+
#   ggtitle("TNIK KO vs WT Selected GO Biological Processes")+
#   theme(axis.text.y=element_text(size=12, face = "bold"),
#         axis.text.x=element_text(size=9, face="bold"),
#         plot.title = element_text(size = 24, face="bold"))
# dev.off()
```

```{r}
res_ko_wt_sig_sort_p <-data.frame(gene = rownames(res_ko_wt), res_ko_wt,
stringsAsFactors = F) %>% 
  na.omit() %>%
  filter(padj<0.001 & log2FoldChange < (-2.0)) %>%
  arrange(padj) %>%
  mutate(minuslogp = -log10(padj), minuslog2p = -log2(padj))

sig_genelist_ko_wt <- res_ko_wt_sig_sort_p$gene

coldata_df <- as.data.frame(coldata)
df <- coldata_df %>% dplyr::select(Sample_Group)

rownames(df) <- row.names(coldata)

df <- df %>% dplyr::arrange(Sample_Group)
df_ko_wt <- df %>% dplyr::filter(Sample_Group %in% c("WT","KO"))

rlogMat_ko_wt <- rlogMat[,row.names(df_ko_wt)]

pdf("Heatmap_siglist_ko_wt.pdf", width=14, height = 32)
pheatmap(subset(rlogMat_ko_wt, rownames(rlogMat_ko_wt) %in% sig_genelist_ko_wt), cluster_rows=TRUE, show_rownames=TRUE,cluster_cols=TRUE, annotation_col=df_ko_wt, scale = "row", fontsize = 20, fontsize_col = 25, fontsize_row = 18, angle_col = 90, main="Genes with LFC <(-2.0) & padj<0.001 in KO vs WT")
dev.off()
```
```{r}
res_ko_wt_sig_sort_p <-data.frame(gene = rownames(res_ko_wt), res_ko_wt,
stringsAsFactors = F) %>% 
  na.omit() %>%
  filter(padj<0.01 & abs(log2FoldChange) > 2.0) %>%
  arrange(padj) %>%
  mutate(minuslogp = -log10(padj), minuslog2p = -log2(padj))

sig_genelist_ko_wt <- res_ko_wt_sig_sort_p$gene

coldata_df <- as.data.frame(coldata)
df <- coldata_df %>% dplyr::select(Sample_Group)

rownames(df) <- row.names(coldata)

df <- df %>% dplyr::arrange(Sample_Group)
df_ko_wt <- df %>% dplyr::filter(Sample_Group %in% c("WT","KO5"))

rlogMat_ko_wt <- rlogMat[,row.names(df_ko_wt)]

pdf("Heatmap_siglist_ko_wt.pdf", width=14, height = 40)
pheatmap(subset(rlogMat_ko_wt, rownames(rlogMat_ko_wt) %in% sig_genelist_ko_wt), cluster_rows=TRUE, show_rownames=TRUE,cluster_cols=TRUE, annotation_col=df_ko_wt, scale = "row", fontsize = 20, fontsize_col = 25, fontsize_row = 18, angle_col = 90, main="Genes with |LFC| > 2.0 & padj<0.01 in KO vs WT")
dev.off()
```
```{r}
res_lof_wt <- DESeq2::results(dds, contrast=c("Sample_Group","LOF1", "WT"), alpha = 0.05)
summary(res_lof_wt)
```

```{r}
lof_wt_list<-data.frame(gene = rownames(res_lof_wt), res_lof_wt,
stringsAsFactors = F) %>% na.omit() %>% dplyr::select("log2FoldChange", "padj") %>% dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp) %>% rownames_to_column(var = "gene")

lof_wt_list <- lof_wt_list %>% 
  dplyr::mutate(padj = ifelse(lof_wt_list$padj == 0, .Machine$double.xmin, padj)) %>%
  dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp)

lof_wt_sorted <- lof_wt_list %>% 
  arrange(desc(abs(LFCminuslogp)))

write.xlsx(lof_wt_sorted, "lof_wt_sorted.xlsx")

pdf(file="Volcano_lof_wt.pdf", width=24, height=21)
ggplot(lof_wt_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2.25,color="grey", alpha=0.5)+
  geom_point(data=subset(lof_wt_list, minuslogp>(-log10(0.001))), size=2.25,color="darkorange1", alpha=1)+
  geom_point(data=subset(lof_wt_list, minuslogp>(-log10(0.001)) & abs(log2FoldChange)>(1)), size=2.25,color="red3", alpha=1)+
  geom_point(data=subset(lof_wt_list, minuslogp<(-log10(0.001)) & abs(log2FoldChange)>(1)), size=2.25,color="yellowgreen", alpha=1)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  ggtitle("TNIK KD LOF vs WT")+
  theme(plot.title = element_text(size = 60, face = "bold"))+
  xlab("Log2FoldChange")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = 1, linetype="dashed")+
  geom_vline(xintercept = -1, linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  geom_label_repel(data=subset(lof_wt_list, minuslogp>(-log10(0.001)) | abs(log2FoldChange)>3), aes(label=gene), size=4, color="black", fontface="bold",  min.segment.length = unit(0, "lines"), max.overlaps = 10)
dev.off()

pdf(file="Volcano_lof_wt_validated.pdf", width=24, height=21)
ggplot(lof_wt_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2.25,color="grey", alpha=0.5)+
  geom_point(data=subset(lof_wt_list, minuslogp>(-log10(0.001))), size=2.25,color="darkorange1", alpha=1)+
  geom_point(data=subset(lof_wt_list, minuslogp>(-log10(0.001)) & abs(log2FoldChange)>(1)), size=2.25,color="red3", alpha=1)+
  geom_point(data=subset(lof_wt_list, minuslogp<(-log10(0.001)) & abs(log2FoldChange)>(1)), size=2.25,color="yellowgreen", alpha=1)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  ggtitle("TNIK KD LOF vs WT")+
  theme(plot.title = element_text(size = 60, face = "bold"))+
  xlab("Log2FoldChange")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = 1, linetype="dashed")+
  geom_vline(xintercept = -1, linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  geom_label_repel(data=subset(lof_wt_list, gene %in% validated_genes), aes(label=gene), size=6, color="black", nudge_y = 0.5, fontface="bold",  min.segment.length = unit(0, "lines"), max.overlaps = 20)
dev.off()
```

```{r}
#make a named vector of P adjusted
lof_wt_gsea <- lof_wt_list$padj 
names(lof_wt_gsea) <- lof_wt_list$gene
ensgEntrez_lof_wt_gsea <- AnnotationDbi::select(org.Hs.eg.db, keys=names(lof_wt_gsea), columns='ENTREZID', keytype='SYMBOL')
df_lof_wt_gsea <- ensgEntrez_lof_wt_gsea %>%
  dplyr::inner_join(lof_wt_list, by=c("SYMBOL"="gene")) %>% 
  dplyr::select(ENTREZID,log2FoldChange,padj) %>% 
  dplyr::group_by(ENTREZID) %>%
  dplyr::summarise(lfc=mean(log2FoldChange), up=min(padj))
lof_wt_lfc<-df_lof_wt_gsea$lfc 
names(lof_wt_lfc)<-df_lof_wt_gsea$ENTREZID

y_lof_wt_reac <- gsePathway(sort((lof_wt_lfc), decreasing=T), organism = "human", pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_lof_wt_reac  <- setReadable(y_lof_wt_reac , 'org.Hs.eg.db', 'ENTREZID')
lof_wt_reac  <- y_lof_wt_reac@result
lof_wt_reac  <- lof_wt_reac  %>% arrange(desc(NES)) %>% mutate(Source = "Reactome")


y_lof_wt_kegg <- gseKEGG(sort((lof_wt_lfc), decreasing=T), organism = "hsa", pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_lof_wt_kegg  <- setReadable(y_lof_wt_kegg , 'org.Hs.eg.db', 'ENTREZID')
lof_wt_kegg  <- y_lof_wt_kegg@result
lof_wt_kegg  <- lof_wt_kegg  %>% arrange(desc(NES)) %>% mutate(Source = "KEGG")


y_lof_wt_go <- gseGO(sort((lof_wt_lfc), decreasing=T), OrgDb=org.Hs.eg.db,
                            ont   = "BP", pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_lof_wt_go <- setReadable(y_lof_wt_go, 'org.Hs.eg.db', 'ENTREZID')
lof_wt_go <- y_lof_wt_go@result
lof_wt_go <- lof_wt_go %>% arrange(desc(NES)) %>% mutate(Source = "GO")

y_lof_wt_msigdb <- GSEA(sort((lof_wt_lfc), decreasing=T), pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, TERM2GENE = msigdb, eps = 1e-100)
y_lof_wt_msigdb <- setReadable(y_lof_wt_msigdb, 'org.Hs.eg.db', 'ENTREZID')
lof_wt_msigdb <- y_lof_wt_msigdb@result
lof_wt_msigdb <- lof_wt_msigdb %>% arrange(desc(NES)) %>% mutate(Source = "MSigDB")


lof_wt_gsea_results <- rbind(lof_wt_go, lof_wt_kegg, lof_wt_msigdb, lof_wt_reac) %>% arrange(NES)
write.xlsx2(lof_wt_gsea_results, "lof_wt_gsea_results.xlsx")

pdf(file="Gsea_volcano_lof_wt.pdf", width=15, height=15)
ggplot(lof_wt_gsea_results, aes(x=NES, y=(-log10(qvalues))))+
  geom_point(aes(size=setSize),color="grey", alpha=0.5)+
  scale_x_continuous(limits = c(-3,3), expand = expansion(mult = 0.5))+
  geom_label_repel(data=subset(lof_wt_gsea_results, NES>1.5 & qvalues < 0.01), aes(label=Description), force_pull=0, force=5, nudge_x = 3, direction = "y", hjust = 0, segment.size = 0.2)+
  geom_label_repel(data=subset(lof_wt_gsea_results, NES<(-1.5) & qvalues < 0.01), aes(label=Description), force=5, nudge_x = -3, hjust = 1, direction = "y", segment.size = 0.2)
dev.off()


msigdb_hallmarks_list_lof_wt <- c("TGFB_UP.V1_UP",
                             "HALLMARK_TGF_BETA_SIGNALING",
                             "HALLMARK_TNFA_SIGNALING_VIA_NFKB")

geneSetID_lof_wt_msigdb = match(msigdb_hallmarks_list_lof_wt,
                              y_lof_wt_msigdb@result[["Description"]])

pdf("gseaplot2_lof_wt_msigdb_hallmarks.pdf", width=12, height=10)
gseaplot2(y_lof_wt_msigdb, geneSetID = geneSetID_lof_wt_msigdb[!is.na(geneSetID_lof_wt_msigdb)], 
          pvalue_table = FALSE, 
          title="TNIK KD LOF vs WT Selected MSigDB Signatures", 
          base_size = 18,
          rel_heights = c(2.5,0.3,0.5))+
  theme(axis.text = element_text(size=12, face = "bold"),
        axis.title = element_text(size=16))
dev.off()


go_list_lof_wt <- c("lymphocyte chemotaxis", "response to interferon-gamma",
                "granulocyte migration", "lymphocyte migration", "positive regulation of leukocyte migration", "neutrophil migration", "eosinophil migration")

pdf("heatplot_lof_wt_msigdb_hallmarks.pdf", width = 60, height=6)
enrichplot::heatplot(y_lof_wt_msigdb, 
                     showCategory = msigdb_hallmarks_list_lof_wt, 
                     foldChange = lof_wt_lfc)+
  ggtitle("TNIK KD LOF vs WT Selected MSigDB Hallmarks")+
  theme(axis.text.y=element_text(size=12, face = "bold"),
        axis.text.x=element_text(size=9, face="bold"),
        plot.title = element_text(size = 24, face="bold"))
dev.off()


pdf("heatplot_lof_wt_go.pdf", width = 20, height=4)
enrichplot::heatplot(y_lof_wt_go, 
                     showCategory = go_list_lof_wt, 
                     foldChange = lof_wt_lfc)+
  ggtitle("TNIK KD LOF vs WT Selected GO Biological Processes")+
  theme(axis.text.y=element_text(size=12, face = "bold"),
        axis.text.x=element_text(size=9, face="bold"),
        plot.title = element_text(size = 24, face="bold"))
dev.off()
```

```{r}
res_ko_lof <- DESeq2::results(dds, contrast=c("Sample_Group","KO5","LOF1"), alpha = 0.05)
summary(res_ko_lof)
```

```{r}
ko_lof_list<-data.frame(gene = rownames(res_ko_lof), res_ko_lof,
stringsAsFactors = F) %>% na.omit() %>% dplyr::select("log2FoldChange", "padj") %>% dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp) %>% rownames_to_column(var = "gene")

ko_lof_list <- ko_lof_list %>% 
  dplyr::mutate(padj = ifelse(ko_lof_list$padj == 0, .Machine$double.xmin, padj)) %>%
  dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp)

ko_lof_sorted <- ko_lof_list %>% 
  arrange(desc(abs(LFCminuslogp)))

write.xlsx(ko_lof_sorted, "ko_lof_sorted.xlsx")

pdf(file="Volcano_ko_lof.pdf", width=24, height=21)
ggplot(ko_lof_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2.25,color="grey", alpha=0.5)+
  geom_point(data=subset(ko_lof_list, minuslogp>(-log10(0.001))), size=2.25,color="darkorange1", alpha=1)+
  geom_point(data=subset(ko_lof_list, minuslogp>(-log10(0.001)) & abs(log2FoldChange)>(1)), size=2.25,color="red3", alpha=1)+
  geom_point(data=subset(ko_lof_list, minuslogp<(-log10(0.001)) & abs(log2FoldChange)>(1)), size=2.25,color="yellowgreen", alpha=1)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  ggtitle("TNIK Full KO vs KD LOF")+
  theme(plot.title = element_text(size = 60, face = "bold"))+
  xlab("Log2FoldChange")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = 1, linetype="dashed")+
  geom_vline(xintercept = -1, linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  geom_label_repel(data=subset(ko_lof_list, minuslogp>(-log10(0.001)) | abs(log2FoldChange)>3), aes(label=gene), size=4, color="black", fontface="bold",  min.segment.length = unit(0, "lines"), max.overlaps = 10)
dev.off()

pdf(file="Volcano_ko_lof_validated.pdf", width=24, height=21)
ggplot(ko_lof_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2.25,color="grey", alpha=0.5)+
  geom_point(data=subset(ko_lof_list, minuslogp>(-log10(0.001))), size=2.25,color="darkorange1", alpha=1)+
  geom_point(data=subset(ko_lof_list, minuslogp>(-log10(0.001)) & abs(log2FoldChange)>(1)), size=2.25,color="red3", alpha=1)+
  geom_point(data=subset(ko_lof_list, minuslogp<(-log10(0.001)) & abs(log2FoldChange)>(1)), size=2.25,color="yellowgreen", alpha=1)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  ggtitle("TNIK Full KO vs KD LOF")+
  theme(plot.title = element_text(size = 60, face = "bold"))+
  xlab("Log2FoldChange")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = 1, linetype="dashed")+
  geom_vline(xintercept = -1, linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  geom_label_repel(data=subset(ko_lof_list, gene %in% validated_genes), aes(label=gene), size=6, color="black", nudge_y = 0.5, fontface="bold",  min.segment.length = unit(0, "lines"), max.overlaps = 20)
dev.off()
```

```{r}
#make a named vector of P adjusted
ko_lof_gsea <- ko_lof_list$padj 
names(ko_lof_gsea) <- ko_lof_list$gene
ensgEntrez_ko_lof_gsea <- AnnotationDbi::select(org.Hs.eg.db, keys=names(ko_lof_gsea), columns='ENTREZID', keytype='SYMBOL')
df_ko_lof_gsea <- ensgEntrez_ko_lof_gsea %>%
  dplyr::inner_join(ko_lof_list, by=c("SYMBOL"="gene")) %>% 
  dplyr::select(ENTREZID,log2FoldChange,padj) %>% 
  dplyr::group_by(ENTREZID) %>%
  dplyr::summarise(lfc=mean(log2FoldChange), up=min(padj))
ko_lof_lfc<-df_ko_lof_gsea$lfc 
names(ko_lof_lfc)<-df_ko_lof_gsea$ENTREZID

y_ko_lof_reac <- gsePathway(sort((ko_lof_lfc), decreasing=T), organism = "human", pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_ko_lof_reac  <- setReadable(y_ko_lof_reac , 'org.Hs.eg.db', 'ENTREZID')
ko_lof_reac  <- y_ko_lof_reac@result
ko_lof_reac  <- ko_lof_reac  %>% arrange(desc(NES)) %>% mutate(Source = "Reactome")


y_ko_lof_kegg <- gseKEGG(sort((ko_lof_lfc), decreasing=T), organism = "hsa", pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_ko_lof_kegg  <- setReadable(y_ko_lof_kegg , 'org.Hs.eg.db', 'ENTREZID')
ko_lof_kegg  <- y_ko_lof_kegg@result
ko_lof_kegg  <- ko_lof_kegg  %>% arrange(desc(NES)) %>% mutate(Source = "KEGG")


y_ko_lof_go <- gseGO(sort((ko_lof_lfc), decreasing=T), OrgDb=org.Hs.eg.db,
                            ont   = "BP", pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_ko_lof_go <- setReadable(y_ko_lof_go, 'org.Hs.eg.db', 'ENTREZID')
ko_lof_go <- y_ko_lof_go@result
ko_lof_go <- ko_lof_go %>% arrange(desc(NES)) %>% mutate(Source = "GO")

y_ko_lof_msigdb <- GSEA(sort((ko_lof_lfc), decreasing=T), pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, TERM2GENE = msigdb, eps = 1e-100)
y_ko_lof_msigdb <- setReadable(y_ko_lof_msigdb, 'org.Hs.eg.db', 'ENTREZID')
ko_lof_msigdb <- y_ko_lof_msigdb@result
ko_lof_msigdb <- ko_lof_msigdb %>% arrange(desc(NES)) %>% mutate(Source = "MSigDB")


ko_lof_gsea_results <- rbind(ko_lof_go, ko_lof_kegg, ko_lof_msigdb, ko_lof_reac) %>% arrange(NES)

pdf(file="Gsea_volcano_ko_lof.pdf", width=15, height=15)
ggplot(ko_lof_gsea_results, aes(x=NES, y=(-log10(qvalues))))+
  geom_point(aes(size=setSize),color="grey", alpha=0.5)+
  scale_x_continuous(limits = c(-3,3), expand = expansion(mult = 0.5))+
  geom_label_repel(data=subset(ko_lof_gsea_results, NES>1.5 & qvalues < 0.01), aes(label=Description), force_pull=0, force=5, nudge_x = 3, direction = "y", hjust = 0, segment.size = 0.2)+
  geom_label_repel(data=subset(ko_lof_gsea_results, NES<(-1.5) & qvalues < 0.01), aes(label=Description), force=5, nudge_x = -3, hjust = 1, direction = "y", segment.size = 0.2)
dev.off()


msigdb_hallmarks_list_ko_lof <- c("HALLMARK_INTERFERON_GAMMA_RESPONSE",
                             "HALLMARK_INTERFERON_ALPHA_RESPONSE",
                             "HALLMARK_TNFA_SIGNALING_VIA_NFKB",
                        "HALLMARK_KRAS_SIGNALING_UP",
                        "HALLMARK_APICAL_JUNCTION",
                       "HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION")

geneSetID_ko_lof_msigdb = match(msigdb_hallmarks_list_ko_lof,
                              y_ko_lof_msigdb@result[["Description"]])

pdf("gseaplot2_ko_wt_msigdb_hallmarks.pdf", width=12, height=10)
gseaplot2(y_ko_lof_msigdb, geneSetID = geneSetID_ko_lof_msigdb[!is.na(geneSetID_ko_lof_msigdb)], 
          pvalue_table = FALSE, 
          title="TNIK KO vs KD LOF Selected MSigDB Signatures", 
          base_size = 18,
          rel_heights = c(2.5,0.3,0.5))+
  theme(axis.text = element_text(size=12, face = "bold"),
        axis.title = element_text(size=16))
dev.off()


go_list_ko_lof <- c("lymphocyte chemotaxis", "response to interferon-gamma",
                "granulocyte migration", "lymphocyte migration", "positive regulation of leukocyte migration", "neutrophil migration", "eosinophil migration")

pdf("heatplot_ko_lof_msigdb_hallmarks.pdf", width = 60, height=6)
enrichplot::heatplot(y_ko_lof_msigdb, 
                     showCategory = msigdb_hallmarks_list_ko_lof, 
                     foldChange = ko_lof_lfc)+
  ggtitle("TNIK KO vs KD LOF Selected MSigDB Hallmarks")+
  theme(axis.text.y=element_text(size=12, face = "bold"),
        axis.text.x=element_text(size=9, face="bold"),
        plot.title = element_text(size = 24, face="bold"))
dev.off()


# pdf("heatplot_ko_lof_go.pdf", width = 20, height=4)
# enrichplot::heatplot(y_ko_lof_go, 
#                      showCategory = go_list_ko_lof, 
#                      foldChange = ko_lof_lfc)+
#   ggtitle("TNIK KO vs KD LOF Selected GO Biological Processes")+
#   theme(axis.text.y=element_text(size=12, face = "bold"),
#         axis.text.x=element_text(size=9, face="bold"),
#         plot.title = element_text(size = 24, face="bold"))
# dev.off()
```
```{r}
ko_lof_gsea_results_sigonly <- ko_lof_gsea_results %>%
  dplyr::filter(qvalues < 0.2)

write.xlsx2(ko_lof_gsea_results_sigonly, "ko_lof_gsea_results.xlsx")


pdf("Gsea_on_volcano_ko_lof.pdf", width=24, height=18)
for(i in 1:nrow(ko_lof_gsea_results_sigonly)){
  ID <- ko_lof_gsea_results_sigonly$ID[i]
  Description <- ko_lof_gsea_results_sigonly$Description[i]
  genelist <- unlist(strsplit(ko_lof_gsea_results_sigonly$core_enrichment[i], "/"))

  print(ggplot(ko_lof_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2.25,color="grey", alpha=0.5)+
  geom_point(data=subset(ko_lof_list, ko_lof_list$gene %in% genelist), color="red3", alpha=1, size=5)+
  theme_bw() +
  ggtitle("TNIK Full KO vs KD LOF")+
  theme(plot.title = element_text(size = 60, face = "bold"))+
  xlab("Log2FoldChange")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = 1, linetype="dashed")+
  geom_vline(xintercept = -1, linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  geom_label_repel(data=subset(ko_lof_list, gene %in% genelist & (ko_lof_list$minuslogp >(-log10(0.01)) | abs(ko_lof_list$log2FoldChange)>3)), aes(label=gene), size=4, color="black", fontface="bold", max.overlaps = Inf)+
  annotate("text", x=4, y=250, label=paste0(Description), size=10, hjust=0, color="red3")+
  annotate("text", x=4, y=240, label=paste0(ID), size=10, hjust=0, color="red3"))}
dev.off()
```

```{r}
a <- ko_wt_list %>% dplyr::mutate(signlogp = (log2FoldChange/abs(log2FoldChange))*minuslogp)
row.names(a) <- a$gene
b <- lof_wt_list %>% dplyr::mutate(signlogp = (log2FoldChange/abs(log2FoldChange))*minuslogp)
row.names(b) <- b$gene

c <- merge(a,b,by=0)
d <- c %>% 
  dplyr::filter(padj.x<0.001, padj.y<0.001)


pdf(file="signlog_ko_lof_purple.pdf", width=15, height=15)
ggplot(c, aes(x=signlogp.x, y=signlogp.y))+
  geom_point(size=2.25,color="grey", alpha=0.6)+
  geom_point(data=subset(c, padj.x>0.001 & padj.y<0.001), size=2.25,color="red", alpha=0.6)+
  geom_point(data=subset(c, padj.x<0.001 & padj.y>0.001), size=2.25,color="blue", alpha=0.6)+
  geom_point(data=d, size=2.25,color="purple", alpha=0.6)+
  #geom_point(data=d, size=2.25,color="red3", alpha=0.6)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  ggtitle("Cross Comparison")+
  theme(plot.title = element_text(size = 30, face = "bold"))+
  #scale_x_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  #scale_y_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  xlab("TNIK KO vs WT (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("TNIK KD LOF vs WT (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = log10(0.001), linetype="dashed")+
  geom_vline(xintercept = -log10(0.001), linetype="dashed")+
  geom_hline(yintercept = log10(0.001), linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  geom_label_repel(data=subset(d, abs(minuslogp.x) > 40 | abs(minuslogp.y) > 40), aes(label=Row.names), size=4, color="purple", fontface="bold", max.overlaps = Inf, min.segment.length = 0, force=5)
dev.off()

pdf(file="LFCcomparison_ko_lof_purple.pdf", width=15, height=15)
ggplot(c, aes(x=log2FoldChange.x, y=log2FoldChange.y))+
  geom_point(size=2.5,color="grey", alpha=0.8)+
  geom_point(data=subset(c, padj.x>0.001 & padj.y<0.001), size=2.25,color="red", alpha=0.6)+
  geom_point(data=subset(c, padj.x<0.001 & padj.y>0.001), size=2.25,color="blue", alpha=0.6)+
  geom_point(data=d, size=2.5,color="purple", alpha=0.6)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) + 
  #xlim(-4,7.5)+
  #ylim(-4,7.5)+
  ggtitle("LFC Comparison at p-adj<0.001 (Cross Comparison)")+
  theme(plot.title = element_text(size = 30, face = "bold"))+
  xlab("TNIK KO vs WT (LFC)")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=25, face = "bold"))+
  ylab("TNIK KD LOF vs WT (LFC)")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=25, face = "bold"))+
  geom_label_repel(data=subset(d, abs(log2FoldChange.x)>2 | abs(log2FoldChange.y)>2), aes(label=Row.names), size=6, color="purple", nudge_y = 0.05, fontface = "bold", min.segment.length = unit(0, "lines"))+
  stat_cor(data=d,method = "spearman", size=4, label.y = 3.5, label.x = -6.35, fontface="italic")+
  annotate("text", x=-5.85, y=4, label="Spearman's Correlation of only\ncommonly significant genes", size = 4, fontface="italic")+
  geom_hline(yintercept = 0, linetype="dashed")+
  geom_vline(xintercept=0, linetype="dashed")
dev.off()

pdf(file="signlog_ko_lof_red.pdf", width=15, height=15)
ggplot(c, aes(x=signlogp.x, y=signlogp.y))+
  geom_point(size=2.25,color="grey", alpha=0.6)+
  geom_point(data=subset(c, padj.x>0.001 & padj.y<0.001), size=2.25,color="red", alpha=0.6)+
  geom_point(data=subset(c, padj.x<0.001 & padj.y>0.001), size=2.25,color="blue", alpha=0.6)+
  geom_point(data=d, size=2.25,color="purple", alpha=0.6)+
  #geom_point(data=d, size=2.25,color="red3", alpha=0.6)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  ggtitle("Cross Comparison")+
  theme(plot.title = element_text(size = 30, face = "bold"))+
  #scale_x_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  #scale_y_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  xlab("TNIK KO vs WT (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("TNIK KD LOF vs WT (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = log10(0.001), linetype="dashed")+
  geom_vline(xintercept = -log10(0.001), linetype="dashed")+
  geom_hline(yintercept = log10(0.001), linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  geom_label_repel(data=subset(c, (padj.x>0.001 & padj.y<0.001) & abs(minuslogp.y) > 30), aes(label=Row.names), size=4, color="red", fontface="bold", max.overlaps = Inf, min.segment.length = 0, force=5)
dev.off()

pdf(file="LFCcomparison_ko_lof_red.pdf", width=15, height=15)
ggplot(c, aes(x=log2FoldChange.x, y=log2FoldChange.y))+
  geom_point(size=2.5,color="grey", alpha=0.8)+
  geom_point(data=subset(c, padj.x>0.001 & padj.y<0.001), size=2.25,color="red", alpha=0.6)+
  geom_point(data=subset(c, padj.x<0.001 & padj.y>0.001), size=2.25,color="blue", alpha=0.6)+
  geom_point(data=d, size=2.5,color="purple", alpha=0.6)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) + 
  #xlim(-4,7.5)+
  #ylim(-4,7.5)+
  ggtitle("LFC Comparison at p-adj<0.001 (Cross Comparison)")+
  theme(plot.title = element_text(size = 30, face = "bold"))+
  xlab("TNIK KO vs WT (LFC)")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=25, face = "bold"))+
  ylab("TNIK KD LOF vs WT (LFC)")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=25, face = "bold"))+
  geom_label_repel(data=subset(c, (padj.x>0.001 & padj.y<0.001) & abs(log2FoldChange.y)>2), aes(label=Row.names), size=6, color="red", nudge_y = 0.05, fontface = "bold", min.segment.length = unit(0, "lines"), max.overlaps = Inf)+
  geom_hline(yintercept = 0, linetype="dashed")+
  geom_vline(xintercept=0, linetype="dashed")
dev.off()

pdf(file="signlog_ko_lof_blue.pdf", width=15, height=15)
ggplot(c, aes(x=signlogp.x, y=signlogp.y))+
  geom_point(size=2.25,color="grey", alpha=0.6)+
  geom_point(data=subset(c, padj.x>0.001 & padj.y<0.001), size=2.25,color="red", alpha=0.6)+
  geom_point(data=subset(c, padj.x<0.001 & padj.y>0.001), size=2.25,color="blue", alpha=0.6)+
  geom_point(data=d, size=2.25,color="purple", alpha=0.6)+
  #geom_point(data=d, size=2.25,color="red3", alpha=0.6)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  ggtitle("Cross Comparison")+
  theme(plot.title = element_text(size = 30, face = "bold"))+
  #scale_x_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  #scale_y_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  xlab("TNIK KO vs WT (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("TNIK KD LOF vs WT (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = log10(0.001), linetype="dashed")+
  geom_vline(xintercept = -log10(0.001), linetype="dashed")+
  geom_hline(yintercept = log10(0.001), linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  geom_label_repel(data=subset(c, (padj.x<0.001 & padj.y>0.001) & abs(minuslogp.x) > 30), aes(label=Row.names), size=4, color="blue", fontface="bold", max.overlaps = Inf, min.segment.length = 0, force=5)
dev.off()

pdf(file="LFCcomparison_ko_lof_blue.pdf", width=15, height=15)
ggplot(c, aes(x=log2FoldChange.x, y=log2FoldChange.y))+
  geom_point(size=2.5,color="grey", alpha=0.8)+
  geom_point(data=subset(c, padj.x>0.001 & padj.y<0.001), size=2.25,color="red", alpha=0.6)+
  geom_point(data=subset(c, padj.x<0.001 & padj.y>0.001), size=2.25,color="blue", alpha=0.6)+
  geom_point(data=d, size=2.5,color="purple", alpha=0.6)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) + 
  #xlim(-4,7.5)+
  #ylim(-4,7.5)+
  ggtitle("LFC Comparison at p-adj<0.001 (Cross Comparison)")+
  theme(plot.title = element_text(size = 30, face = "bold"))+
  xlab("TNIK KO vs WT (LFC)")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=25, face = "bold"))+
  ylab("TNIK KD LOF vs WT (LFC)")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=25, face = "bold"))+
  geom_label_repel(data=subset(c, (padj.x<0.001 & padj.y>0.001) & abs(log2FoldChange.x)>2), aes(label=Row.names), size=6, color="blue", nudge_y = 0.05, fontface = "bold", min.segment.length = unit(0, "lines"), max.overlaps = Inf)+
  geom_hline(yintercept = 0, linetype="dashed")+
  geom_vline(xintercept=0, linetype="dashed")
dev.off()
```
```{r}
common_gsea <- merge(ko_wt_gsea_results, lof_wt_gsea_results, by="ID", all=TRUE)
for (i in 1:nrow(common_gsea)){
  if(is.na(common_gsea$Description.x[i])){
    common_gsea$Description.x[i] <- common_gsea$Description.y[i]
    common_gsea$qvalues.x[i] <- 1}
  if(is.na(common_gsea$Description.y[i])){
    common_gsea$Description.y[i] <- common_gsea$Description.x[i]
    common_gsea$qvalues.y[i] <- 1}}
common_gsea <- common_gsea %>% mutate(Significance = ifelse(common_gsea$qvalues.x<=0.1 & common_gsea$qvalues.y<=0.1, "Both", ifelse(common_gsea$qvalues.x<=0.1 & common_gsea$qvalues.y>0.1, "Only KO", ifelse(common_gsea$qvalues.x>0.1 & common_gsea$qvalues.y<=0.1, "Only KD LOF", "Neither"))))
common_gsea$qval_maximum = rowMaxs(as.matrix(common_gsea[,c(8,18)]))
common_gsea <- common_gsea %>%
    dplyr::arrange(qval_maximum)
common_gsea_Excel <- common_gsea %>% 
  dplyr::select(-qval_maximum, -Description.y, -pvalue.x, -pvalue.y, -p.adjust.x, -p.adjust.y, -Source.x) %>%
  dplyr::rename("Description" = "Description.x", "qval KO" = "qvalues.x", "qval KD LOF" = "qvalues.y", "CoreEnrichment (KO vs Control)" = "core_enrichment.x", "CoreEnrichment (KD LOF vs Control)" = "core_enrichment.y","setSize(KOvsControl)" =  "setSize.x", "setSize(KDLOFvsControl)" =  "setSize.y", "enrichmentScore(KOvsControl)" =  "enrichmentScore.x",  "enrichmentScore(KDLOFvsControl)" =  "enrichmentScore.y", "NES(KOvsControl)" =  "NES.x", "NES(KDLOFvsControl)" =  "NES.y", "LeadingEdge(KOvsControl" = "leading_edge.x", "LeadingEdge(KDLOFvsControl" = "leading_edge.y", "rank(KOvsControl)" = "rank.x", "rank(KDLOFvsControl)" = "rank.y", "Source" = "Source.y") 

write.xlsx(common_gsea_Excel, "GSEA Common.xlsx")
```


```{r}
pdf("sig_enrich_nes_ko_lof.pdf", width=12, height=8)
ggplot(common_gsea, aes(x=NES.x, y=NES.y))+
  geom_point(colour = "grey", alpha = 0.5)+
  geom_point(data=subset(common_gsea,Significance=="Both"),
             colour = "purple",size = 2,alpha = 0.7) +
  geom_point(data=subset(common_gsea,Significance=="Only KD LOF"),
             colour = "blue", size = 2,alpha = 0.7) +
  geom_point(data=subset(common_gsea,Significance=="Only KO"),
             colour = "red", size = 2,alpha = 0.7) +
  geom_label_repel(data = subset(common_gsea, (abs(NES.x) > 1.8 | abs(NES.y) >1.8) & Significance %in% c("Both")), aes(label=Description.x), size=2, color="black", fontface="bold",  min.segment.length = unit(0, "lines"), max.overlaps = Inf, force = 5)+
  xlim(-4,4)+
  ylim(-4,4)+
  xlab("NES of KO vs WT")+
  ylab("NES of KD LOF vs WT")+
  theme_bw()
dev.off()


pdf("sig_enrich_p_ko_lof.pdf", width=12, height=8)
ggplot(common_gsea, aes(x=(-log10(qvalues.x)*(NES.x/abs(NES.x))), y=(-log10(qvalues.y)*(NES.y/abs(NES.y)))))+
  geom_point(colour = "grey", alpha = 0.5)+
  geom_point(data=subset(common_gsea,Significance=="Both"),
             colour = "purple",size = 2,alpha = 0.7) +
  geom_point(data=subset(common_gsea,Significance=="Only KD LOF"),
             colour = "blue", size = 2,alpha = 0.7) +
  geom_point(data=subset(common_gsea,Significance=="Only KO"),
             colour = "red", size = 2,alpha = 0.7) +
  geom_label_repel(data = subset(common_gsea, Significance %in% c("Both")), aes(label=Description.x), size=2, color="black", fontface="bold",  min.segment.length = unit(1, "lines"), max.overlaps = 30)+
  xlab("Signed qval of KO vs WT")+
  ylab("Signed qval of KD LOF vs WT")+
  theme_bw()
dev.off()
```
```{r}
a <- ko_wt_list %>% dplyr::mutate(signlogp = (log2FoldChange/abs(log2FoldChange))*minuslogp)
row.names(a) <- a$gene
b <- lof_wt_list %>% dplyr::mutate(signlogp = (log2FoldChange/abs(log2FoldChange))*minuslogp)
row.names(b) <- b$gene

c <- merge(a,b,by=0)
d <- c %>% 
  dplyr::filter(padj.x<0.001, padj.y<0.001)

common_gsea_Excel_sigonly <- common_gsea_Excel %>%
  dplyr::filter(!common_gsea_Excel$Significance == "Neither")

pdf("Gsea_on_signlog_ko_lof.pdf", width=24, height=18)
for(i in 1:nrow(common_gsea_Excel_sigonly)){
  ID <- common_gsea_Excel_sigonly$ID[i]
  Description <- common_gsea_Excel_sigonly$Description[i]
  genelist1 <- unlist(strsplit(common_gsea_Excel_sigonly$`CoreEnrichment (KO vs Control)`[i], "/"))
  genelist2 <- unlist(strsplit(common_gsea_Excel_sigonly$`CoreEnrichment (KD LOF vs Control)`[i], "/"))
  genelist <- unique(c(genelist1,genelist2))


  print(ggplot(c, aes(x=signlogp.x, y=signlogp.y))+
  geom_point(size=2.25,color="grey", alpha=0.6)+
  geom_point(data=subset(c, c$Row.names %in% genelist), color="red3", alpha=1, size=5)+
  theme_bw() +
  #scale_x_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  #scale_y_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  xlab("TNIK KO vs WT (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("TNIK KD LOF vs WT (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = log10(0.001), linetype="dashed")+
  geom_vline(xintercept = -log10(0.001), linetype="dashed")+
  geom_hline(yintercept = log10(0.001), linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  geom_label_repel(data=subset(c, Row.names %in% genelist & (c$minuslogp.x >(-log10(0.001)) | c$minuslogp.y > (-log10(0.001)))), aes(label=Row.names), size=5, color="black", fontface="bold", max.overlaps = Inf)+
  annotate("text", x=-150, y=200, label=paste0(Description), size=10, hjust=0, color="red3")+
  annotate("text", x=-150, y=180, label=paste0(ID), size=10, hjust=0, color="red3"))}
dev.off()


pdf("Gsea_on_lfc_ko_lof.pdf", width=24, height=18)
for(i in 1:nrow(common_gsea_Excel_sigonly)){
  ID <- common_gsea_Excel_sigonly$ID[i]
  Description <- common_gsea_Excel_sigonly$Description[i]
  genelist1 <- unlist(strsplit(common_gsea_Excel_sigonly$`CoreEnrichment (KO vs Control)`[i], "/"))
  genelist2 <- unlist(strsplit(common_gsea_Excel_sigonly$`CoreEnrichment (KD LOF vs Control)`[i], "/"))
  genelist <- unique(c(genelist1,genelist2))


  print(ggplot(c, aes(x=log2FoldChange.x, y=log2FoldChange.y))+
  geom_point(size=2.5,color="grey", alpha=0.8)+
  geom_point(data=subset(c, c$Row.names %in% genelist), color="red3", alpha=1, size=5)+
  theme_bw() +
  xlab("TNIK KO vs WT (LFC)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=25, face = "bold"))+
  ylab("TNIK KD LOF vs WT (LFC)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=25, face = "bold"))+
  geom_hline(yintercept = 0, linetype="dashed")+
  geom_vline(xintercept=0, linetype="dashed")+
  geom_label_repel(data=subset(c, Row.names %in% genelist & (abs(c$log2FoldChange.x) >2 | abs(c$log2FoldChange.y) >2)), aes(label=Row.names), size=5, color="black", fontface="bold", max.overlaps = Inf)+
  annotate("text", x=-6, y=5, label=paste0(Description), size=10, hjust=0, color="red3")+
  annotate("text", x=-6, y=4, label=paste0(ID), size=10, hjust=0, color="red3"))}
dev.off()
```

```{r}
res_wttgfb_wt <- DESeq2::results(dds, contrast=c("Sample_Group","WTTGFB","WT"), alpha = 0.05)
summary(res_wttgfb_wt)
```
```{r}
wttgfb_wt_list<-data.frame(gene = rownames(res_wttgfb_wt), res_wttgfb_wt,
stringsAsFactors = F) %>% na.omit() %>% dplyr::select("log2FoldChange", "padj") %>% dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp) %>% rownames_to_column(var = "gene")

wttgfb_wt_list <- wttgfb_wt_list %>% 
  dplyr::mutate(padj = ifelse(wttgfb_wt_list$padj == 0, .Machine$double.xmin, padj)) %>%
  dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp)

wttgfb_wt_sorted <- wttgfb_wt_list %>% 
  arrange(desc(abs(LFCminuslogp)))

write.xlsx(wttgfb_wt_sorted, "wttgfb_wt_sorted.xlsx")

pdf(file="Volcano_wttgfb_wt.pdf", width=24, height=21)
ggplot(wttgfb_wt_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2.25,color="grey", alpha=0.5)+
  geom_point(data=subset(wttgfb_wt_list, minuslogp>(-log10(0.001))), size=2.25,color="darkorange1", alpha=1)+
  geom_point(data=subset(wttgfb_wt_list, minuslogp>(-log10(0.001)) & abs(log2FoldChange)>(1)), size=2.25,color="red3", alpha=1)+
  geom_point(data=subset(wttgfb_wt_list, minuslogp<(-log10(0.001)) & abs(log2FoldChange)>(1)), size=2.25,color="yellowgreen", alpha=1)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  ggtitle("TNIK WT+TGFB vs WT")+
  theme(plot.title = element_text(size = 60, face = "bold"))+
  xlab("Log2FoldChange")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = 1, linetype="dashed")+
  geom_vline(xintercept = -1, linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  geom_label_repel(data=subset(wttgfb_wt_list, minuslogp>(-log10(0.001)) | abs(log2FoldChange)>3), aes(label=gene), size=7, color="black", fontface="bold",  min.segment.length = unit(0, "lines"), max.overlaps = 10)
dev.off()

pdf(file="Volcano_wttgfb_wt_validated.pdf", width=24, height=21)
ggplot(wttgfb_wt_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=2.25,color="grey", alpha=0.5)+
  geom_point(data=subset(wttgfb_wt_list, minuslogp>(-log10(0.001))), size=2.25,color="darkorange1", alpha=1)+
  geom_point(data=subset(wttgfb_wt_list, minuslogp>(-log10(0.001)) & abs(log2FoldChange)>(1)), size=2.25,color="red3", alpha=1)+
  geom_point(data=subset(wttgfb_wt_list, minuslogp<(-log10(0.001)) & abs(log2FoldChange)>(1)), size=2.25,color="yellowgreen", alpha=1)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  ggtitle("TNIK WT+TGFB vs WT")+
  theme(plot.title = element_text(size = 60, face = "bold"))+
  xlab("Log2FoldChange")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = 1, linetype="dashed")+
  geom_vline(xintercept = -1, linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  geom_label_repel(data=subset(wttgfb_wt_list, gene %in% validated_genes), aes(label=gene), size=6, color="black", nudge_y = 0.5, fontface="bold",  min.segment.length = unit(0, "lines"), max.overlaps = 20)
dev.off()
```

```{r}
wttgfb_wt_gsea <- wttgfb_wt_list$padj
names(wttgfb_wt_gsea) <- wttgfb_wt_list$gene
ensgEntrez_wttgfb_wt_gsea <- AnnotationDbi::select(org.Hs.eg.db, keys=names(wttgfb_wt_gsea), columns='ENTREZID', keytype='SYMBOL')
df_wttgfb_wt_gsea <- ensgEntrez_wttgfb_wt_gsea %>%
  dplyr::inner_join(wttgfb_wt_list, by=c("SYMBOL"="gene")) %>% 
  dplyr::select(ENTREZID,log2FoldChange,padj) %>% 
  dplyr::group_by(ENTREZID) %>%
  dplyr::summarise(lfc=mean(log2FoldChange), up=min(padj))
wttgfb_wt_lfc<-df_wttgfb_wt_gsea$lfc 
names(wttgfb_wt_lfc)<-df_wttgfb_wt_gsea$ENTREZID

y_wttgfb_wt_reac <- gsePathway(sort((wttgfb_wt_lfc), decreasing=T), organism = "human", pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_wttgfb_wt_reac  <- setReadable(y_wttgfb_wt_reac , 'org.Hs.eg.db', 'ENTREZID')
wttgfb_wt_reac  <- y_wttgfb_wt_reac@result
wttgfb_wt_reac  <- wttgfb_wt_reac  %>% arrange(desc(NES)) %>% mutate(Source = "Reactome")


y_wttgfb_wt_kegg <- gseKEGG(sort((wttgfb_wt_lfc), decreasing=T), organism = "hsa", pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_wttgfb_wt_kegg  <- setReadable(y_wttgfb_wt_kegg , 'org.Hs.eg.db', 'ENTREZID')
wttgfb_wt_kegg  <- y_wttgfb_wt_kegg@result
wttgfb_wt_kegg  <- wttgfb_wt_kegg  %>% arrange(desc(NES)) %>% mutate(Source = "KEGG")


y_wttgfb_wt_go <- gseGO(sort((wttgfb_wt_lfc), decreasing=T), OrgDb=org.Hs.eg.db,
                            ont   = "BP", pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_wttgfb_wt_go <- setReadable(y_wttgfb_wt_go, 'org.Hs.eg.db', 'ENTREZID')
wttgfb_wt_go <- y_wttgfb_wt_go@result
wttgfb_wt_go <- wttgfb_wt_go %>% arrange(desc(NES)) %>% mutate(Source = "GO")

y_wttgfb_wt_msigdb <- GSEA(sort((wttgfb_wt_lfc), decreasing=T), pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, TERM2GENE = msigdb, eps = 1e-100)
y_wttgfb_wt_msigdb <- setReadable(y_wttgfb_wt_msigdb, 'org.Hs.eg.db', 'ENTREZID')
wttgfb_wt_msigdb <- y_wttgfb_wt_msigdb@result
wttgfb_wt_msigdb <- wttgfb_wt_msigdb %>% arrange(desc(NES)) %>% mutate(Source = "MSigDB")


wttgfb_wt_gsea_results <- rbind(wttgfb_wt_go, wttgfb_wt_kegg, wttgfb_wt_msigdb, wttgfb_wt_reac) %>% arrange(desc(NES))
write.xlsx2(wttgfb_wt_gsea_results, "wttgfb_wt_gsea_results.xlsx")



msigdb_hallmarks_list_wttgfb_wt <- c("HALLMARK_TNFA_SIGNALING_VIA_NFKB",
                             "TGFB_UP.V1_UP",
                             "HALLMARK_TGF_BETA_SIGNALING")

geneSetID_wttgfb_wt_msigdb = match(msigdb_hallmarks_list_wttgfb_wt,
                              y_wttgfb_wt_msigdb@result[["Description"]])

pdf("gseaplot2_wttgfb_wt_tgfb_sigs.pdf", width=12, height=10)
gseaplot2(y_wttgfb_wt_msigdb, geneSetID = geneSetID_wttgfb_wt_msigdb[!is.na(geneSetID_wttgfb_wt_msigdb)], 
          pvalue_table = FALSE, 
          title="WT+TGFB vs WT MSigDB Signatures", 
          base_size = 18,
          rel_heights = c(2.5,0.3,0.5))+
  theme(axis.text = element_text(size=12, face = "bold"),
        axis.title = element_text(size=16))
dev.off()
```


```{r}
res_kotgfb_ko <- DESeq2::results(dds, contrast=c("Sample_Group","KO5TGFB","KO5"), alpha = 0.001)
summary(res_kotgfb_ko)
```

```{r}
kotgfb_ko_list<-data.frame(gene = rownames(res_kotgfb_ko), res_kotgfb_ko,
stringsAsFactors = F) %>% na.omit() %>% dplyr::select("log2FoldChange", "padj") %>% dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp) %>% rownames_to_column(var = "gene")

kotgfb_ko_list <- kotgfb_ko_list %>% 
  dplyr::mutate(padj = ifelse(kotgfb_ko_list$padj == 0, .Machine$double.xmin, padj)) %>%
  dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp)

kotgfb_ko_gsea <- kotgfb_ko_list$padj 
names(kotgfb_ko_gsea) <- kotgfb_ko_list$gene
ensgEntrez_kotgfb_ko_gsea <- AnnotationDbi::select(org.Hs.eg.db, keys=names(kotgfb_ko_gsea), columns='ENTREZID', keytype='SYMBOL')
df_kotgfb_ko_gsea <- ensgEntrez_kotgfb_ko_gsea %>%
  dplyr::inner_join(kotgfb_ko_list, by=c("SYMBOL"="gene")) %>% 
  dplyr::select(ENTREZID,log2FoldChange,padj) %>% 
  dplyr::group_by(ENTREZID) %>%
  dplyr::summarise(lfc=mean(log2FoldChange), up=min(padj))
kotgfb_ko_lfc<-df_kotgfb_ko_gsea$lfc 
names(kotgfb_ko_lfc)<-df_kotgfb_ko_gsea$ENTREZID

y_kotgfb_ko_reac <- gsePathway(sort((kotgfb_ko_lfc), decreasing=T), organism = "human", pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_kotgfb_ko_reac  <- setReadable(y_kotgfb_ko_reac , 'org.Hs.eg.db', 'ENTREZID')
kotgfb_ko_reac  <- y_kotgfb_ko_reac@result
kotgfb_ko_reac  <- kotgfb_ko_reac  %>% arrange(desc(NES)) %>% mutate(Source = "Reactome")


y_kotgfb_ko_kegg <- gseKEGG(sort((kotgfb_ko_lfc), decreasing=T), organism = "hsa", pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_kotgfb_ko_kegg  <- setReadable(y_kotgfb_ko_kegg , 'org.Hs.eg.db', 'ENTREZID')
kotgfb_ko_kegg  <- y_kotgfb_ko_kegg@result
kotgfb_ko_kegg  <- kotgfb_ko_kegg  %>% arrange(desc(NES)) %>% mutate(Source = "KEGG")


y_kotgfb_ko_go <- gseGO(sort((kotgfb_ko_lfc), decreasing=T), OrgDb=org.Hs.eg.db,
                            ont   = "BP", pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_kotgfb_ko_go <- setReadable(y_kotgfb_ko_go, 'org.Hs.eg.db', 'ENTREZID')
kotgfb_ko_go <- y_kotgfb_ko_go@result
kotgfb_ko_go <- kotgfb_ko_go %>% arrange(desc(NES)) %>% mutate(Source = "GO")

y_kotgfb_ko_msigdb <- GSEA(sort((kotgfb_ko_lfc), decreasing=T), pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, TERM2GENE = msigdb, eps = 1e-100)
y_kotgfb_ko_msigdb <- setReadable(y_kotgfb_ko_msigdb, 'org.Hs.eg.db', 'ENTREZID')
kotgfb_ko_msigdb <- y_kotgfb_ko_msigdb@result
kotgfb_ko_msigdb <- kotgfb_ko_msigdb %>% arrange(desc(NES)) %>% mutate(Source = "MSigDB")


kotgfb_ko_gsea_results <- rbind(kotgfb_ko_go, kotgfb_ko_kegg, kotgfb_ko_msigdb, kotgfb_ko_reac) %>% arrange(NES)
write.xlsx2(kotgfb_ko_gsea_results, "kotgfb_ko_gsea_results.xlsx")

msigdb_hallmarks_list_kotgfb_ko <- c("HALLMARK_TNFA_SIGNALING_VIA_NFKB",
                             "TGFB_UP.V1_UP",
                             "HALLMARK_TGF_BETA_SIGNALING")

geneSetID_kotgfb_ko_msigdb = match(msigdb_hallmarks_list_kotgfb_ko,
                              y_kotgfb_ko_msigdb@result[["Description"]])

pdf("gseaplot2_kotgfb_ko_tgfb_sigs.pdf", width=12, height=10)
gseaplot2(y_kotgfb_ko_msigdb, geneSetID = geneSetID_kotgfb_ko_msigdb[!is.na(geneSetID_kotgfb_ko_msigdb)], 
          pvalue_table = FALSE, 
          title="KO+TGFB vs KO MSigDB Signatures", 
          base_size = 18,
          rel_heights = c(2.5,0.3,0.5))+
  theme(axis.text = element_text(size=12, face = "bold"),
        axis.title = element_text(size=16))
dev.off()
```

```{r}
res_loftgfb_lof <- DESeq2::results(dds, contrast=c("Sample_Group","LOF1TGFB","LOF1"), alpha = 0.001)
summary(res_loftgfb_lof)
```

```{r}
loftgfb_lof_list<-data.frame(gene = rownames(res_loftgfb_lof), res_loftgfb_lof,
stringsAsFactors = F) %>% na.omit() %>% dplyr::select("log2FoldChange", "padj") %>% dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp) %>% rownames_to_column(var = "gene")

loftgfb_lof_list <- loftgfb_lof_list %>% 
  dplyr::mutate(padj = ifelse(loftgfb_lof_list$padj == 0, .Machine$double.xmin, padj)) %>%
  dplyr::mutate(minuslogp = -log10(padj), "LFCminuslogp" = log2FoldChange*minuslogp)

loftgfb_lof_gsea <- loftgfb_lof_list$padj 
names(loftgfb_lof_gsea) <- loftgfb_lof_list$gene
ensgEntrez_loftgfb_lof_gsea <- AnnotationDbi::select(org.Hs.eg.db, keys=names(loftgfb_lof_gsea), columns='ENTREZID', keytype='SYMBOL')
df_loftgfb_lof_gsea <- ensgEntrez_loftgfb_lof_gsea %>%
  dplyr::inner_join(loftgfb_lof_list, by=c("SYMBOL"="gene")) %>% 
  dplyr::select(ENTREZID,log2FoldChange,padj) %>% 
  dplyr::group_by(ENTREZID) %>%
  dplyr::summarise(lfc=mean(log2FoldChange), up=min(padj))
loftgfb_lof_lfc<-df_loftgfb_lof_gsea$lfc 
names(loftgfb_lof_lfc)<-df_loftgfb_lof_gsea$ENTREZID

y_loftgfb_lof_reac <- gsePathway(sort((loftgfb_lof_lfc), decreasing=T), organism = "human", pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_loftgfb_lof_reac  <- setReadable(y_loftgfb_lof_reac , 'org.Hs.eg.db', 'ENTREZID')
loftgfb_lof_reac  <- y_loftgfb_lof_reac@result
loftgfb_lof_reac  <- loftgfb_lof_reac  %>% arrange(desc(NES)) %>% mutate(Source = "Reactome")


y_loftgfb_lof_kegg <- gseKEGG(sort((loftgfb_lof_lfc), decreasing=T), organism = "hsa", pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_loftgfb_lof_kegg  <- setReadable(y_loftgfb_lof_kegg , 'org.Hs.eg.db', 'ENTREZID')
loftgfb_lof_kegg  <- y_loftgfb_lof_kegg@result
loftgfb_lof_kegg  <- loftgfb_lof_kegg  %>% arrange(desc(NES)) %>% mutate(Source = "KEGG")


y_loftgfb_lof_go <- gseGO(sort((loftgfb_lof_lfc), decreasing=T), OrgDb=org.Hs.eg.db,
                            ont   = "BP", pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, eps = 1e-100)
y_loftgfb_lof_go <- setReadable(y_loftgfb_lof_go, 'org.Hs.eg.db', 'ENTREZID')
loftgfb_lof_go <- y_loftgfb_lof_go@result
loftgfb_lof_go <- loftgfb_lof_go %>% arrange(desc(NES)) %>% mutate(Source = "GO")

y_loftgfb_lof_msigdb <- GSEA(sort((loftgfb_lof_lfc), decreasing=T), pvalueCutoff=0.1, pAdjustMethod="BH", verbose=F, TERM2GENE = msigdb, eps = 1e-100)
y_loftgfb_lof_msigdb <- setReadable(y_loftgfb_lof_msigdb, 'org.Hs.eg.db', 'ENTREZID')
loftgfb_lof_msigdb <- y_loftgfb_lof_msigdb@result
loftgfb_lof_msigdb <- loftgfb_lof_msigdb %>% arrange(desc(NES)) %>% mutate(Source = "MSigDB")


loftgfb_lof_gsea_results <- rbind(loftgfb_lof_go, loftgfb_lof_kegg, loftgfb_lof_msigdb, loftgfb_lof_reac) %>% arrange(NES)
write.xlsx2(loftgfb_lof_gsea_results, "loftgfb_lof_gsea_results.xlsx")

msigdb_hallmarks_list_loftgfb_lof <- c("HALLMARK_TNFA_SIGNALING_VIA_NFKB",
                             "TGFB_UP.V1_UP",
                             "HALLMARK_TGF_BETA_SIGNALING")

geneSetID_loftgfb_lof_msigdb = match(msigdb_hallmarks_list_loftgfb_lof,
                              y_loftgfb_lof_msigdb@result[["Description"]])

pdf("gseaplot2_loftgfb_lof_tgfb_sigs.pdf", width=12, height=10)
gseaplot2(y_loftgfb_lof_msigdb, geneSetID = geneSetID_loftgfb_lof_msigdb[!is.na(geneSetID_loftgfb_lof_msigdb)], 
          pvalue_table = FALSE, 
          title="LOF+TGFB vs LOF MSigDB Signatures", 
          base_size = 18,
          rel_heights = c(2.5,0.3,0.5))+
  theme(axis.text = element_text(size=12, face = "bold"),
        axis.title = element_text(size=16))
dev.off()
```

```{r}
a <- kotgfb_ko_list %>% dplyr::mutate(signlogp = (log2FoldChange/abs(log2FoldChange))*minuslogp)
row.names(a) <- a$gene
b <- loftgfb_lof_list %>% dplyr::mutate(signlogp = (log2FoldChange/abs(log2FoldChange))*minuslogp)
row.names(b) <- b$gene

c <- merge(a,b,by=0)
d <- c %>% 
  dplyr::filter(padj.x<0.001, padj.y<0.001)

pdf(file="signlog_kotgfb_loftgfb_purple.pdf", width=15, height=15)
ggplot(c, aes(x=signlogp.x, y=signlogp.y))+
  geom_point(size=2.25,color="grey", alpha=0.6)+
  geom_point(data=subset(c, padj.x>0.001 & padj.y<0.001), size=2.25,color="red", alpha=0.6)+
  geom_point(data=subset(c, padj.x<0.001 & padj.y>0.001), size=2.25,color="blue", alpha=0.6)+
  geom_point(data=d, size=2.25,color="purple", alpha=0.6)+
  #geom_point(data=d, size=2.25,color="red3", alpha=0.6)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  ggtitle("Cross Comparison")+
  theme(plot.title = element_text(size = 30, face = "bold"))+
  #scale_x_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  #scale_y_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  xlab("TNIK KO+TGFB vs KO (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("TNIK KD LOF+TGFB vs KD LOF (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = log10(0.001), linetype="dashed")+
  geom_vline(xintercept = -log10(0.001), linetype="dashed")+
  geom_hline(yintercept = log10(0.001), linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  geom_label_repel(data=d, aes(label=Row.names), size=4, color="purple", fontface="bold", max.overlaps = Inf, min.segment.length = 0, force=5)
dev.off()

pdf(file="LFCcomparison_kotgfb_loftgfb_purple.pdf", width=15, height=15)
ggplot(c, aes(x=log2FoldChange.x, y=log2FoldChange.y))+
  geom_point(size=2.5,color="grey", alpha=0.8)+
  geom_point(data=subset(c, padj.x>0.001 & padj.y<0.001), size=2.25,color="red", alpha=0.6)+
  geom_point(data=subset(c, padj.x<0.001 & padj.y>0.001), size=2.25,color="blue", alpha=0.6)+
  geom_point(data=d, size=2.5,color="purple", alpha=0.6)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) + 
  #xlim(-4,7.5)+
  #ylim(-4,7.5)+
  ggtitle("LFC Comparison at p-adj<0.001 (Cross Comparison)")+
  theme(plot.title = element_text(size = 30, face = "bold"))+
  xlab("TNIK KO+TGFB vs KO (LFC)")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=25, face = "bold"))+
  ylab("TNIK KD LOF+TGFB vs KD LOF (LFC)")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=25, face = "bold"))+
  geom_label_repel(data=d, aes(label=Row.names), size=6,color="purple",nudge_y = 0.5, nudge_x = 0.5, fontface = "bold", max.overlaps = Inf, force = 5, min.segment.length = 0)+
  stat_cor(data=d,method = "spearman", size=6, label.y = 3, label.x = -2.55, fontface="italic", vjust=0)+
  annotate("text", x=-1.85, y=3.5, label="Spearman's Correlation of only\ncommonly significant genes", size = 6, fontface="italic", vjust=0)+
  geom_hline(yintercept = 0, linetype="dashed")+
  geom_vline(xintercept=0, linetype="dashed")
dev.off()

pdf(file="signlog_kotgfb_loftgfb_red.pdf", width=15, height=15)
ggplot(c, aes(x=signlogp.x, y=signlogp.y))+
  geom_point(size=2.25,color="grey", alpha=0.6)+
  geom_point(data=subset(c, padj.x>0.001 & padj.y<0.001), size=2.25,color="red", alpha=0.6)+
  geom_point(data=subset(c, padj.x<0.001 & padj.y>0.001), size=2.25,color="blue", alpha=0.6)+
  geom_point(data=d, size=2.25,color="purple", alpha=0.6)+
  #geom_point(data=d, size=2.25,color="red3", alpha=0.6)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  ggtitle("Cross Comparison")+
  theme(plot.title = element_text(size = 30, face = "bold"))+
  #scale_x_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  #scale_y_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  xlab("TNIK KO+TGFB vs KO (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("TNIK KD LOF+TGFB vs KD LOF (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = log10(0.001), linetype="dashed")+
  geom_vline(xintercept = -log10(0.001), linetype="dashed")+
  geom_hline(yintercept = log10(0.001), linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  geom_label_repel(data=subset(c, padj.x>0.001 & padj.y<0.001), aes(label=Row.names), size=4, color="red", fontface="bold", max.overlaps = Inf, min.segment.length = 0, force=5)
dev.off()

pdf(file="LFCcomparison_kotgfb_loftgfb_red.pdf", width=15, height=15)
ggplot(c, aes(x=log2FoldChange.x, y=log2FoldChange.y))+
  geom_point(size=2.5,color="grey", alpha=0.8)+
  geom_point(data=subset(c, padj.x>0.001 & padj.y<0.001), size=2.25,color="red", alpha=0.6)+
  geom_point(data=subset(c, padj.x<0.001 & padj.y>0.001), size=2.25,color="blue", alpha=0.6)+
  geom_point(data=d, size=2.5,color="purple", alpha=0.6)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) + 
  ggtitle("LFC Comparison at p-adj<0.001 (Cross Comparison)")+
  theme(plot.title = element_text(size = 30, face = "bold"))+
  xlab("TNIK KO+TGFB vs KO (LFC)")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=25, face = "bold"))+
  ylab("TNIK KD LOF+TGFB vs KD LOF (LFC)")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=25, face = "bold"))+
  geom_label_repel(data=subset(c, padj.x>0.001 & padj.y<0.001), aes(label=Row.names), size=6,color="red",nudge_y = 0.5, nudge_x = 0.5, fontface = "bold", max.overlaps = Inf, force = 5, min.segment.length = 0)+
  geom_hline(yintercept = 0, linetype="dashed")+
  geom_vline(xintercept=0, linetype="dashed")
dev.off()

pdf(file="signlog_kotgfb_loftgfb_blue.pdf", width=15, height=15)
ggplot(c, aes(x=signlogp.x, y=signlogp.y))+
  geom_point(size=2.25,color="grey", alpha=0.6)+
  geom_point(data=subset(c, padj.x>0.001 & padj.y<0.001), size=2.25,color="red", alpha=0.6)+
  geom_point(data=subset(c, padj.x<0.001 & padj.y>0.001), size=2.25,color="blue", alpha=0.6)+
  geom_point(data=d, size=2.25,color="purple", alpha=0.6)+
  #geom_point(data=d, size=2.25,color="red3", alpha=0.6)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) +
  ggtitle("Cross Comparison")+
  theme(plot.title = element_text(size = 30, face = "bold"))+
  #scale_x_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  #scale_y_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  xlab("TNIK KO+TGFB vs KO (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("TNIK KD LOF+TGFB vs KD LOF (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = log10(0.001), linetype="dashed")+
  geom_vline(xintercept = -log10(0.001), linetype="dashed")+
  geom_hline(yintercept = log10(0.001), linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  geom_label_repel(data=subset(c, -log10(padj.x)>10 & padj.y>0.001), aes(label=Row.names), size=4, color="blue", fontface="bold", max.overlaps = Inf, min.segment.length = 0, force=5)
dev.off()

pdf(file="LFCcomparison_kotgfb_loftgfb_blue.pdf", width=15, height=15)
ggplot(c, aes(x=log2FoldChange.x, y=log2FoldChange.y))+
  geom_point(size=2.5,color="grey", alpha=0.8)+
  geom_point(data=subset(c, padj.x>0.001 & padj.y<0.001), size=2.25,color="red", alpha=0.6)+
  geom_point(data=subset(c, padj.x<0.001 & padj.y>0.001), size=2.25,color="blue", alpha=0.6)+
  geom_point(data=d, size=2.5,color="purple", alpha=0.6)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(),plot.background = element_blank(), axis.line = element_line()) + 
  ggtitle("LFC Comparison at p-adj<0.001 (Cross Comparison)")+
  theme(plot.title = element_text(size = 30, face = "bold"))+
  xlab("TNIK KO+TGFB vs KO (LFC)")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=25, face = "bold"))+
  ylab("TNIK KD LOF+TGFB vs KD LOF (LFC)")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=25, face = "bold"))+
  geom_label_repel(data=subset(c, (padj.x<0.001 & padj.y>0.001) & abs(log2FoldChange.x) >1), aes(label=Row.names), size=6,color="blue",nudge_y = 0.5, nudge_x = 0.5, fontface = "bold", max.overlaps = Inf, force = 5, min.segment.length = 0)+
  geom_hline(yintercept = 0, linetype="dashed")+
  geom_vline(xintercept=0, linetype="dashed")
dev.off()
```
```

```

```{r}
common_gsea_tgfb <- merge(kotgfb_ko_gsea_results, loftgfb_lof_gsea_results, by="ID", all=TRUE)
for (i in 1:nrow(common_gsea_tgfb)){
  if(is.na(common_gsea_tgfb$Description.x[i])){
    common_gsea_tgfb$Description.x[i] <- common_gsea_tgfb$Description.y[i]
    common_gsea_tgfb$qvalues.x[i] <- 1}
  if(is.na(common_gsea_tgfb$Description.y[i])){
    common_gsea_tgfb$Description.y[i] <- common_gsea_tgfb$Description.x[i]
    common_gsea_tgfb$qvalues.y[i] <- 1}}
common_gsea_tgfb <- common_gsea_tgfb %>% mutate(Significance = ifelse(common_gsea_tgfb$qvalues.x<=0.1 & common_gsea_tgfb$qvalues.y<=0.1, "Both", ifelse(common_gsea_tgfb$qvalues.x<=0.1 & common_gsea_tgfb$qvalues.y>0.1, "Only KO+TGFB", ifelse(common_gsea_tgfb$qvalues.x>0.1 & common_gsea_tgfb$qvalues.y<=0.1, "Only KD LOF+TGFB", "Neither"))))
common_gsea_tgfb$qval_maximum = rowMaxs(as.matrix(common_gsea_tgfb[,c(8,18)]))
common_gsea_tgfb <- common_gsea_tgfb %>%
    dplyr::arrange(qval_maximum)
common_gsea_tgfb_Excel <- common_gsea_tgfb %>% 
  dplyr::select(-qval_maximum, -Description.y, -pvalue.x, -pvalue.y, -p.adjust.x, -p.adjust.y, -Source.x) %>%
  dplyr::rename("Description" = "Description.x", "p-adj KO" = "qvalues.x", "p-adj KD LOF" = "qvalues.y", "CoreEnrichment (KO+TFGB vs KO)" = "core_enrichment.x", "CoreEnrichment (KD LOF+TGFB vs KD LOF)" = "core_enrichment.y","setSize(KO+TFGB vs KO)" =  "setSize.x", "setSize(KD LOF+TGFB vs KD LOF)" =  "setSize.y", "enrichmentScore(KO+TFGB vs KO)" =  "enrichmentScore.x",  "enrichmentScore(KD LOF+TGFB vs KD LOF)" =  "enrichmentScore.y", "NES(KO+TFGB vs KO)" =  "NES.x", "NES(KD LOF+TGFB vs KD LOF)" =  "NES.y", "LeadingEdge(KO+TFGB vs KO" = "leading_edge.x", "LeadingEdge(KD LOF+TGFB vs KD LOF" = "leading_edge.y", "rank(KO+TFGB vs KO)" = "rank.x", "rank(KD LOF+TGFB vs KD LOF)" = "rank.y", "Source" = "Source.y") 

write.xlsx(common_gsea_tgfb_Excel, "GSEA Common TGFB.xlsx")
```


```{r}
pdf("sig_enrich_nes_tgfb.pdf", width=12, height=8)
ggplot(common_gsea_tgfb, aes(x=NES.x, y=NES.y))+
  geom_point(colour = "grey", alpha = 0.5)+
  geom_point(data=subset(common_gsea_tgfb,Significance=="Both"),
             colour = "purple",size = 2,alpha = 0.7) +
  geom_point(data=subset(common_gsea_tgfb,Significance=="Only KD LOF+TGFB"),
             colour = "blue", size = 2,alpha = 0.7) +
  geom_point(data=subset(common_gsea_tgfb,Significance=="Only KO+TGFB"),
             colour = "red", size = 2,alpha = 0.7) +
  geom_label_repel(data = subset(common_gsea_tgfb, Significance %in% c("Both")), aes(label=Description.x), size=2, color="black", fontface="bold",  min.segment.length = unit(1, "lines"), max.overlaps = 30)+
  xlim(-4,4)+
  ylim(-4,4)+
  xlab("NES of KO+TGFB vs KO")+
  ylab("NES of KD LOF+TGFB vs KD LOF")+
  theme_bw()
dev.off()


pdf("sig_enrich_p_tgfb.pdf", width=12, height=8)
ggplot(common_gsea_tgfb, aes(x=(-log10(qvalues.x)*(NES.x/abs(NES.x))), y=(-log10(qvalues.y)*(NES.y/abs(NES.y)))))+
  geom_point(colour = "grey", alpha = 0.5)+
  geom_point(data=subset(common_gsea_tgfb,Significance=="Both"),
             colour = "purple",size = 2,alpha = 0.7) +
  geom_point(data=subset(common_gsea_tgfb,Significance=="Only KD LOF+TGFB"),
             colour = "blue", size = 2,alpha = 0.7) +
  geom_point(data=subset(common_gsea_tgfb,Significance=="Only KO+TGFB"),
             colour = "red", size = 2,alpha = 0.7) +
  geom_label_repel(data = subset(common_gsea_tgfb, Significance %in% c("Both")), aes(label=Description.x), size=2, color="black", fontface="bold",  min.segment.length = unit(1, "lines"), max.overlaps = 30)+
  xlab("Signed qval of KO+TGFB vs KO")+
  ylab("Signed qval of KD LOF+TGFB vs KD LOF")+
  theme_bw()
dev.off()
```

```{r}
a <- kotgfb_ko_list %>% dplyr::mutate(signlogp = (log2FoldChange/abs(log2FoldChange))*minuslogp)
row.names(a) <- a$gene
b <- loftgfb_lof_list %>% dplyr::mutate(signlogp = (log2FoldChange/abs(log2FoldChange))*minuslogp)
row.names(b) <- b$gene

c <- merge(a,b,by=0)
d <- c %>% 
  dplyr::filter(padj.x<0.001, padj.y<0.001)

common_gsea_Excel_tgfb_sigonly <- common_gsea_tgfb_Excel %>%
  dplyr::filter(!common_gsea_tgfb_Excel$Significance == "Neither")

pdf("Gsea_on_signlog_tgfb.pdf", width=24, height=18)
for(i in 1:nrow(common_gsea_Excel_tgfb_sigonly)){
  ID <- common_gsea_Excel_tgfb_sigonly$ID[i]
  Description <- common_gsea_Excel_tgfb_sigonly$Description[i]
  genelist1 <- unlist(strsplit(common_gsea_Excel_tgfb_sigonly$`CoreEnrichment (KO+TFGB vs KO)`[i], "/"))
  genelist2 <- unlist(strsplit(common_gsea_Excel_tgfb_sigonly$`CoreEnrichment (KD LOF+TGFB vs KD LOF)`[i], "/"))
  genelist <- unique(c(genelist1,genelist2))


  print(ggplot(c, aes(x=signlogp.x, y=signlogp.y))+
  geom_point(size=2.25,color="grey", alpha=0.6)+
  geom_point(data=subset(c, c$Row.names %in% genelist), color="red3", alpha=1, size=5)+
  theme_bw() +
  #scale_x_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  #scale_y_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  xlab("TNIK KO+TGFB vs KO (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("TNIK KD LOF+TGFB vs KD LOF (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = log10(0.001), linetype="dashed")+
  geom_vline(xintercept = -log10(0.001), linetype="dashed")+
  geom_hline(yintercept = log10(0.001), linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  geom_label_repel(data=subset(c, Row.names %in% genelist & (c$minuslogp.x >(-log10(0.001)) | c$minuslogp.y > (-log10(0.001)))), aes(label=Row.names), size=5, color="black", fontface="bold", max.overlaps = Inf)+
  annotate("text", x=-150, y=200, label=paste0(Description), size=10, hjust=0, color="red3")+
  annotate("text", x=-150, y=180, label=paste0(ID), size=10, hjust=0, color="red3"))}
dev.off()


pdf("Gsea_on_lfc_tgfb.pdf", width=24, height=18)
for(i in 1:nrow(common_gsea_Excel_tgfb_sigonly)){
  ID <- common_gsea_Excel_tgfb_sigonly$ID[i]
  Description <- common_gsea_Excel_tgfb_sigonly$Description[i]
  genelist1 <- unlist(strsplit(common_gsea_Excel_tgfb_sigonly$`CoreEnrichment (KO+TFGB vs KO)`[i], "/"))
  genelist2 <- unlist(strsplit(common_gsea_Excel_tgfb_sigonly$`CoreEnrichment (KD LOF+TGFB vs KD LOF)`[i], "/"))
  genelist <- unique(c(genelist1,genelist2))


  print(ggplot(c, aes(x=log2FoldChange.x, y=log2FoldChange.y))+
  geom_point(size=2.5,color="grey", alpha=0.8)+
  geom_point(data=subset(c, c$Row.names %in% genelist), color="red3", alpha=1, size=5)+
  theme_bw() +
  #xlim(-4,7.5)+
  #ylim(-4,7.5)+
  xlab("TNIK KO+TGFB vs KO (LFC)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=25, face = "bold"))+
  ylab("TNIK KD LOF+TGFB vs KD LOF (LFC)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=25, face = "bold"))+
  geom_hline(yintercept = 0, linetype="dashed")+
  geom_vline(xintercept=0, linetype="dashed")+
  geom_label_repel(data=subset(c, Row.names %in% genelist & (abs(c$log2FoldChange.x) >2 | abs(c$log2FoldChange.y) >2)), aes(label=Row.names), size=5, color="black", fontface="bold", max.overlaps = Inf)+
  annotate("text", x=-6, y=5, label=paste0(Description), size=10, hjust=0, color="red3")+
  annotate("text", x=-6, y=4, label=paste0(ID), size=10, hjust=0, color="red3"))}
dev.off()
```


```{r}
wttgfb_nfkb <- wttgfb_wt_gsea_results %>%
  dplyr::filter(Description == "HALLMARK_TNFA_SIGNALING_VIA_NFKB")
kotgfb_nfkb <- kotgfb_ko_gsea_results %>%
  dplyr::filter(Description == "HALLMARK_TNFA_SIGNALING_VIA_NFKB")
loftgfb_nfkb <- loftgfb_lof_gsea_results %>%
  dplyr::filter(Description == "HALLMARK_TNFA_SIGNALING_VIA_NFKB")


genelist_tnfa_nfkb_1 <- unlist(strsplit(wttgfb_nfkb$core_enrichment, "/"))
genelist_tnfa_nfkb_2 <- unlist(strsplit(kotgfb_nfkb$core_enrichment, "/"))
genelist_tnfa_nfkb_3 <- unlist(strsplit(loftgfb_nfkb$core_enrichment, "/"))
genelist_tnfa_nfkb_union <- unique(c(genelist_tnfa_nfkb_1, genelist_tnfa_nfkb_2, genelist_tnfa_nfkb_3))

wttgfb_wt_list <- wttgfb_wt_list %>% mutate(SampleGroup = "WT")
kotgfb_ko_list <- kotgfb_ko_list %>% mutate(SampleGroup = "KO")
loftgfb_lof_list <- loftgfb_lof_list %>% mutate(SampleGroup = "LOF")

boxplot_data <- rbind(wttgfb_wt_list, kotgfb_ko_list, loftgfb_lof_list)
boxplot_data_nfkb <- boxplot_data %>%
  dplyr::filter(gene %in% genelist_tnfa_nfkb_union)

ggplot(boxplot_data_nfkb, aes(x=SampleGroup, y=log2FoldChange, fill=SampleGroup))+
  geom_boxplot(outlier.shape = NA)+
  geom_jitter(width=0.2, alpha=0.6)


merged_wt_ko_tgfb <- merge(wttgfb_wt_list, kotgfb_ko_list, by="gene")
boxplot_data_merged_wt_ko_tgfb <- merged_wt_ko_tgfb %>%
    dplyr::filter(gene %in% genelist_tnfa_nfkb_union)

ggpaired(boxplot_data_merged_wt_ko_tgfb, cond1="log2FoldChange.x", 
         cond2="log2FoldChange.y", color = "condition", label = "gene", repel = TRUE, palette = "jco") +
  stat_compare_means(label = "p.format", paired = TRUE, method = "wilcox.test",
                     label.x.npc = "right")
```

```{r}
wttgfb_tgfb <- wttgfb_wt_gsea_results %>%
  dplyr::filter(Description == "HALLMARK_TGF_BETA_SIGNALING")
kotgfb_tgfb <- kotgfb_ko_gsea_results %>%
  dplyr::filter(Description == "HALLMARK_TGF_BETA_SIGNALING")
loftgfb_tgfb <- loftgfb_lof_gsea_results %>%
  dplyr::filter(Description == "HALLMARK_TGF_BETA_SIGNALING")


genelist_tgfb_1 <- unlist(strsplit(wttgfb_tgfb$core_enrichment, "/"))
genelist_tgfb_2 <- unlist(strsplit(kotgfb_tgfb$core_enrichment, "/"))
genelist_tgfb_3 <- unlist(strsplit(loftgfb_tgfb$core_enrichment, "/"))
genelist_tgfb_union <- unique(c(genelist_tgfb_1, genelist_tgfb_2, genelist_tgfb_3))

msigdb_hallmark_tgfb <- msigdb %>% dplyr::filter(gs_name == "HALLMARK_TGF_BETA_SIGNALING")

tgfb_symbols <- bitr(msigdb_hallmark_tgfb$entrez_gene, fromType="ENTREZID", toType="SYMBOL", OrgDb="org.Hs.eg.db")


boxplot_data_merged_wt_ko_tgfb <- merged_wt_ko_tgfb %>%
  dplyr::filter((gene %in% tgfb_symbols$SYMBOL) & (padj.x<1 & padj.y <1))

pdf("tgfb_hallmark_wt_ko.pdf", width=6, height=8)
ggpaired(boxplot_data_merged_wt_ko_tgfb, cond1="log2FoldChange.x", 
         cond2="log2FoldChange.y", color = "condition", label = "gene", repel = TRUE, palette = "jco") +
  stat_compare_means(label = "p.format", paired = TRUE, method = "wilcox.test",
                     label.x.npc = "right")
dev.off()

merged_wt_lof_tgfb <- merge(wttgfb_wt_list, loftgfb_lof_list, by="gene")
boxplot_data_merged_wt_lof_tgfb <- merged_wt_lof_tgfb %>%
  dplyr::filter((gene %in% tgfb_symbols$SYMBOL) & (padj.x<1 & padj.y <1))

pdf("tgfb_hallmark_wt_lof.pdf", width=6, height=8)
ggpaired(boxplot_data_merged_wt_lof_tgfb, cond1="log2FoldChange.x", 
         cond2="log2FoldChange.y", color = "condition", label = "gene", repel = TRUE, palette = "jco") +
  stat_compare_means(label = "p.format", paired = TRUE, method = "wilcox.test",
                     label.x.npc = "right")
dev.off()
```

```{r}
ko_wt_list <- ko_wt_list %>%
  arrange(LFCminuslogp)

pdf(file="Pub_Volcano_ko_wt.pdf", width=18, height=18)
ggplot(ko_wt_list, aes(x=log2FoldChange, y=(minuslogp)))+
  geom_point(size=5,color="grey", alpha=0.5)+
  geom_point(data=ko_wt_list[1:30,], size=10, color="red3")+
  theme_bw()+
  ggtitle("TNIK Full KO vs WT")+
  theme(plot.title = element_text(size = 60, face = "bold"))+
  xlab("Log2FoldChange")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("-log10(p-adj)")+
  theme(axis.title = element_text(size = 40, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_label_repel(data=ko_wt_list[1:30,], aes(label=gene), size=8, color="black", fontface="bold",  min.segment.length = unit(0, "lines"), max.overlaps = Inf, nudge_x=-10, direction = "y", hjust = 1, force=5)
dev.off()
```

```{r}
a <- ko_wt_list %>% dplyr::mutate(signlogp = (log2FoldChange/abs(log2FoldChange))*minuslogp)
row.names(a) <- a$gene
b <- lof_wt_list %>% dplyr::mutate(signlogp = (log2FoldChange/abs(log2FoldChange))*minuslogp)
row.names(b) <- b$gene

c <- merge(a,b,by=0)
d <- c %>% 
  dplyr::filter(padj.x<0.001, padj.y<0.001)

pdf(file="Pub_signlog_ko_lof.pdf", width=10, height=10)
ggplot(c, aes(x=signlogp.x, y=signlogp.y))+
  geom_point(size=4,color="grey60", alpha=0.7)+
  geom_point(data=d, size=4,color="red3", alpha=0.5)+
  theme_bw() +
  ggtitle("Cross Comparison")+
  theme(plot.title = element_text(size = 30, face = "bold"))+
  #scale_x_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  #scale_y_continuous(limits = c(-200,200),  breaks = seq(-200, 200, by = 20))+
  xlab("TNIK KO vs WT (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  ylab("TNIK KD LOF vs WT (sign(LFC) x -log10(p-adj))")+
  theme(axis.title = element_text(size = 20, face = "bold"), axis.text=element_text(size=20, face = "bold"))+
  geom_vline(xintercept = log10(0.001), linetype="dashed")+
  geom_vline(xintercept = -log10(0.001), linetype="dashed")+
  geom_hline(yintercept = log10(0.001), linetype="dashed")+
  geom_hline(yintercept = -log10(0.001), linetype="dashed")+
  stat_cor(data=d, method = "spearman", size=8, label.y = 200, label.x = -150)+
  geom_smooth(method='lm', formula= y~x, color="blue4" ,alpha=1, size=4, span=0.3)
  #geom_label_repel(data=subset(d, abs(minuslogp.x) > 50 & abs(minuslogp.y) > 50), aes(label=Row.names), size=4, fontface="bold", max.overlaps = Inf, min.segment.length = 0, force=5)
dev.off()
```

